src/burn/engine/plan.cpp

```diff
@@ -575,7 +595,11 @@ extern "C" HRESULT PlanRegistration(
     UINT cDependencies = 0;
 
     pPlan->fRegister = TRUE; // register the bundle since we're modifying machine state.
-    pPlan->fKeepRegistrationDefault = (pRegistration->fInstalled || BOOTSTRAPPER_RESUME_TYPE_REBOOT == resumeType); // keep the registration if the bundle was already installed or we are planning after a restart.
+
+    // Keep the registration if the bundle was already installed or we are planning after a restart.
+    // Also keep the registration if we are caching because the ARP (via uninstall) allows uncaching.
+    pPlan->fKeepRegistrationDefault = (pRegistration->fInstalled || BOOTSTRAPPER_RESUME_TYPE_REBOOT == resumeType || BOOTSTRAPPER_ACTION_CACHE == pPlan->action);
```

Comment:
Looking through this further, I think this was originally since caching was a background process (which I also removed). Perhaps something to reconsider if/when we add async caching back in.